### ISSUE TITLE
Some partial updates are not reflected to active node

### ISSUE THREAD

**Issue by riida** [vespa-engine/vespa#9826]
Hi, We are using vespa7 configured with redundancy 2 and searchable-copies 2.
We found a problem that some latest partial update with document api (JSON REST API) are not reflected to the active node.
These update seems to be reflected to only non-active node, but these bucket timestamps in two nodes are the same.
(Get and Visit operations return old updated data about specific field.)

```
$ vespa-stat --document id:dynamic:dynamic::123.456
Bucket maps to the following actual files:
        BucketInfo(BucketId(0x4000000000002c84): [distributor:7] [node(idx=7,crc=0x6bb28b86,docs=392/517,bytes=1443203/1448753,trusted=true,active=true,ready=true), node(idx=2,crc=0x6bb28b86,docs=392/517,bytes=1496407/1501957,trusted=true,active=false,ready=true)])
 
Details for BucketId(0x4000000000002c84):
        Bucket information from node 2:
Persistence bucket BucketId(0x4000000000002c84), partition 0
  Timestamp: 1560188135000055, Doc(id:dynamic:dynamic::123.456), gid(0x842ca07a720d52d081e07aac), size: 3973
 
 
        Bucket information from node 7:
Persistence bucket BucketId(0x4000000000002c84), partition 0
  Timestamp: 1560188135000055, Doc(id:dynamic:dynamic::123.456), gid(0x842ca07a720d52d081e07aac), size: 2820
```
I mean, each nodes received exactly same gid, bucketid and timestamp, but one node's data size is different from the other.

Is there any solution to reflect latest partial update to active node certainly?

Vespa version: 7.51.13-1

Thanks.

**Comment by vekterli** [comment]
In a system where bucket replicas are in sync all documents should have the same content across all replicas, so this is a bug.

Are the fields you're observing divergence in _attributes_? I.e. are they marked as `indexing: attribute` in the search definition? We've very recently uncovered and fixed a bug related to background compaction where certain partial updates to attributes could be lost (#9630). The fix for this is available in 7.53.50 and beyond, so if this is the case I'd strongly recommend upgrading and refeeding the affected documents.

If the affected fields are not attributes it's a thus far unknown bug. Any information you might have regarding what events could have transpired in the cluster around (or before) the time of the last document operation (Mon Jun 10 17:35:35 UTC 2019) would be greatly appreciated.
* Any emitted node events or cluster state changes by the `container-clustercontroller`service?
* Are there any entries in the vespa logs on the content nodes matching the pattern `updated documents with different timestamps`? This warning will be printed if a distributor believes all bucket replicas to be in sync prior to an operation, only to discover after the fact that it wasn't.

Thank you for reporting this!

**Comment by riida** [comment]
Thank you for your quick reply.

These fields are _attributes_ fields as you said.
So we upgrade vespa version to 7.58.16-1 and refeed documents, the problem seems to be resolved.

Thank you for giving me good advice!
